CollapseLauncher/Classes/Helper/Image/ImageLoaderHelper.cs

```diff
@@ -20,25 +19,50 @@
 using Windows.Storage;
 using Windows.Storage.Streams;
 using Orientation = Microsoft.UI.Xaml.Controls.Orientation;
+using static CollapseLauncher.Helper.Image.Waifu2X;
+using static Hi3Helper.Shared.Region.LauncherConfig;
 
 namespace CollapseLauncher.Helper.Image
 {
     internal static class ImageLoaderHelper
     {
-        internal static Dictionary<string, string> SupportedImageFormats = 
+        internal static Dictionary<string, string> SupportedImageFormats =
             new() { { "Supported formats", "*.jpg;*.jpeg;*.jfif;*.png;*.bmp;*.tiff;*.tif;*.webp" } };
 
+        private static Waifu2X _waifu2X;
+
+        public static Waifu2XStatus Waifu2XStatus => _waifu2X.Status;
+
+        public static bool IsWaifu2XEnabled
+        {
+            get => GetAppConfigValue("EnableWaifu2X").ToBool() && IsWaifu2XUsable;
+            set => SetAndSaveConfigValue("EnableWaifu2X", value);
+        }
+
+        public static bool IsWaifu2XUsable => Waifu2XStatus != Waifu2XStatus.NotAvailable && Waifu2XStatus != Waifu2XStatus.TestNotPassed;
+
+        public static void InitWaifu2X()
+        {
+            _waifu2X = new Waifu2X();
+            if (_waifu2X.Status == Waifu2XStatus.NotAvailable)
+                return;
+            _waifu2X.Set(Param.Noise, -1);
+            _waifu2X.Set(Param.Scale, 2);
+            _waifu2X.Load(Path.Combine(AppFolder, @"Assets\Models\scale2.0x_model.param.bin"),
+                Path.Combine(AppFolder, @"Assets\Models\scale2.0x_model.bin"));
+        }
+
         internal static async Task<FileStream> LoadImage(string path, bool isUseImageCropper = false, bool overwriteCachedImage = false)
         {
             if (string.IsNullOrEmpty(path) || !File.Exists(path)) return null;
             double aspectRatioX = InnerLauncherConfig.m_actualMainFrameSize.Width;
             double aspectRatioY = InnerLauncherConfig.m_actualMainFrameSize.Height;
             double dpiScale = InnerLauncherConfig.m_appDPIScale;
-            uint targetSourceImageWidth = (uint)(aspectRatioX * 1.5 * dpiScale);
-            uint targetSourceImageHeight = (uint)(aspectRatioY * 1.5 * dpiScale);
+            uint targetSourceImageWidth = (uint)(aspectRatioX * dpiScale);
```

Comment:
IIRC, Waifu2X works best when its used to upscale the image to 2x the size (probably an old understanding of mine, I did not follow the development quite much).
If its the case then can you try seeing iif scaling it 2x and let WASDK scale the image down themself will bring better (sharper) image or not?

Reply:
waifu2x can only upscale a image to 2x. What you described is exactly the current approach: first scale to 2x using waifu2x (only once) and then scale down (maybe still up) using MagicScaler.

This change is not related to waifu2x. I think we can get a good enough image with MagicScaler, so the additional `* 1.5` is not needed anymore. Introduced in d9e6725.

Reply:
got it~